src/lib/components/json-schema/MessageInputSwitch.tsx

```diff
@@ -0,0 +1,85 @@
+import { Flex } from "@chakra-ui/react";
+import type { Dispatch, SetStateAction } from "react";
+import { useRef } from "react";
+
+import { Tooltip } from "../Tooltip";
+import { MotionBox } from "lib/components/MotionBox";
+
+export enum MessageTabs {
+  JSON_INPUT = "JSON Input",
+  YOUR_SCHEMA = "Your Schema",
+}
+
+export const jsonInputFormKey = MessageTabs.JSON_INPUT as "JSON Input";
+export const yourSchemaInputFormKey = MessageTabs.YOUR_SCHEMA as "Your Schema";
+
+interface MessageInputSwitchProps {
+  currentTab: MessageTabs;
+  disabled?: boolean;
+  onTabChange: Dispatch<SetStateAction<MessageTabs>>;
+}
+
+const tabs = Object.values(MessageTabs);
+
+export const MessageInputSwitch = ({
+  currentTab,
+  disabled = false,
+  onTabChange,
+}: MessageInputSwitchProps) => {
+  const tabRefs = useRef<(HTMLDivElement | null)[]>([]);
+  const activeIndex = tabs.indexOf(currentTab);
+  return (
+    <Tooltip label="Select or fill code id first" isDisabled={!disabled}>
+      <div>
```

Comment:
remove?

Reply:
not sure why but if there's no div the tooltip somewhat lost reference to the component (not showing up)

Reply:
Ah,I forgot about Tooltip problem.